internal/corazawaf/rule.go

```diff
@@ -85,6 +85,23 @@ func (p *inferredPhases) has(phase types.RulePhase) bool {
 	return (*p & (1 << phase)) != 0
 }
 
+// hasOrMinor returns true if the phase is set or any phase before it
+// E.g.
+// inferredPhases = 00000010 (types.PhaseRequestHeaders)
+// hasOrMinor(types.PhaseRequestBody) performs:
+// 00000010 & 00000001
+// 00000010 & 00000010
+// 00000010 & 00000100
+// If any of the them is true, it returns true and stops iterating
+func (p *inferredPhases) hasOrMinor(phase types.RulePhase) bool {
+	for i := 1; i <= int(phase); i++ {
```

Comment:
The first idea was relying on a mask (looking at the example like 00000111), but I found that the code was becoming more cryptic, with supported variables. I went for the small loop.
Happy if any better implementation can be done without too many byzantine